src/Test/Spec/Reporter/TC.purs

```diff
@@ -0,0 +1,99 @@
+-- | Team City reporter, also the one used for intellij
+module Test.Spec.Reporter.TC (tcReporter, teamcity) where
+
+import Prelude
+
+import Control.Monad.State (get, put)
+import Data.Array ((:))
+import Data.Array (intercalate, last) as Array
+import Data.Int (trunc)
+import Data.Maybe (fromMaybe) as Maybe
+import Data.String as String
+import Data.Time.Duration (Milliseconds(..))
+import Data.Tuple (Tuple)
+import Data.Tuple.Nested ((/\))
+import Test.Spec.Console (tellLn)
+import Test.Spec.Reporter.Base (defaultReporter)
+import Test.Spec.Result (Result(..))
+import Test.Spec.Runner (Reporter)
+import Test.Spec.Runner.Event (Event(..)) as Event
+import Test.Spec.Tree as Tree
+
+nameFromPath :: Tree.Path -> String
+nameFromPath path = Tree.parentSuiteName path
+    # Array.last
+    # Maybe.fromMaybe ""
+
+escape :: String -> String
+escape = String.replaceAll
+  (String.Pattern "|")
+  (String.Replacement "||")
+  >>> String.replaceAll
+  (String.Pattern "\n")
+  (String.Replacement "|n")
+  >>> String.replaceAll
+    (String.Pattern "'")
+    (String.Replacement "|'")
+  >>> String.replaceAll
+    (String.Pattern "\r")
+    (String.Replacement "|r")
+  >>> String.replaceAll
+    (String.Pattern "[")
+    (String.Replacement "|[")
+  >>> String.replaceAll
+    (String.Pattern "]")
+    (String.Replacement "|]")
```

Comment:
May I suggest even indentation for easier reading?

```suggestion
escape = 
  String.replaceAll
    (String.Pattern "|")
    (String.Replacement "||")
  >>> String.replaceAll
    (String.Pattern "\n")
    (String.Replacement "|n")
  >>> String.replaceAll
    (String.Pattern "'")
    (String.Replacement "|'")
  >>> String.replaceAll
    (String.Pattern "\r")
    (String.Replacement "|r")
  >>> String.replaceAll
    (String.Pattern "[")
    (String.Replacement "|[")
  >>> String.replaceAll
    (String.Pattern "]")
    (String.Replacement "|]")
```

Reply:
Or, perhaps, even a more concise way:

```suggestion
escape =
  replace "|" "||"
  >>> replace "\n" "|n"
  >>> replace "'" "|'"
  >>> replace "\r" "|r"
  >>> replace "[" "|["
  >>> replace "]" "|]"
  where
    replace x y = String.replaceAll (String.Pattern x) (String.Replacement y)
```

Reply:
I think that maybe if going this far, using regexp would even be better